### DISCUSSION TITLE
validate if [card] payment method already exists using fingerprint

### DISCUSSION THREAD

**Discussion by fernan542** [flutter-stripe/flutter_stripe#691] [Q&A] [answered]
**Describe the bug**
How can I validate if card already exists or attached to customer? Because as of now, the customer can add many Cards with the same details. I've read in some SDKs / Library that we can check it using the `payment_method`'s fingerprint but I'm not able to see it when I access the `Card` properties.

Card properties:
```
/// Card data associated with the payment method
class Card with _$Card {
  @JsonSerializable(explicitToJson: true)
  const factory Card({
    /// The brand associated to the card e.g. (visa, amex).
    String? brand,

    /// Two letter iso code.
    String? country,

    /// four digit number representing the year of expiry of the card.
    int? expYear,

    /// two digit number representing the month of expire of the card.
    int? expMonth,

    /// card funding type e.g. (credit, debit).
    String? funding,

    /// last four digits of the card.
    String? last4,
  }) = _Card;

  factory Card.fromJson(Map<String, dynamic> json) => _$CardFromJson(json);
}
```

 - Package version: 2.4.0
 - Flutter version 2.10.3

My temporary solution is to use the `last4` property of the given card.

**Comment by jonasbark** [comment, marked as answer]
Please add more details on your use case - are you using payment sheets? Or the CardField?
This is likely something where the Stripe support can help you.

**Reply by fernan542** [reply]
Sorry for not detailed use case.

The scenario is:
1. The customer adds payment method through the provided function:
```
  final pm = await Stripe.instance
          .createPaymentMethod(const PaymentMethodParams.card());
```
Let's say, the details of the card was the testing card (424242...).

2. If the creation was successful, the app will send the `payment_method_id` and `customer_id` to the Go server in order to attach them because the Stripe's Flutter SDK doesn't support it yet.

3. Then the app updates the UI that contains the newly added payment method. (assuming that it doesn't have dupe)

4. The customer can still add other `payment_method`. And here comes the problem, since Stripe allows us to create a `payment_method` even it contains the same details (4242424....), it will be saved to the Stripe server and if the app requests the List of Payment Methods, we can see that it contains 2 identical `payment_method` (42424242...).

Based on the other Libraries, if the model `PaymentMethod` was called, it contains a `fingerprint` no matter what type it is. But on Stripe's Flutter SDK, the `fingerprint` can only be used on selected types.

**Reply by remonh87** [reply]
yes you are correct in this. We use the official react native sdk native mappers for our data objects and stripe has implemented the fingerprint for SEPA, Becs, AUBECS and usbankaccount. Feel free to create a ticket in the[ react native repo](https://github.com/stripe/stripe-react-native/issues) and we will take it along once they implemented it.

**Reply by fernan542** [reply]
Ok sirs, thank you very much for your responses. I've created a ticket: https://github.com/stripe/stripe-react-native/issues/907

**Reply by fernan542** [reply]
**Update**
This issue was fixed on Stripe's react native repo. https://github.com/stripe/stripe-react-native/pull/914

**Reply by remonh87** [reply]
@fernan542  thanks for the heads up. It missed release v0.8.0 of react native but we will take it along when Stripe releases the next version.

**Reply by fernan542** [reply]
Oki doks. As long as there's no reported problem with this scenario in our app, since I'm only using the `last4` as the validation, I can wait but a little bit risky. Hopefully I can immediately implement those changes at least this month.

Many thanks!

**Reply by Flucadetena** [reply]
Hi is this getting updated any time soon? We were using the "fingerprint" from the js api and updating to new things in the Flutter plugin we can not make use of the fingerprint no more.

**Reply by Flucadetena** [reply]
I created a pull request as  saw it was an easy fix. Hope it helps an this issue can be closed.

Thanks for the time and work.

**Reply by Flucadetena** [reply]
Hi @fernan542 not sure if you still need this but I created a pull request to fix this issue. Hopefully it will get merged soon ;)
In the meantime you can use my fork like this. Hope it helps ;)

In your pubspec.yaml
```
  flutter_stripe:
    git:
      url: https://github.com/Flucadetena/flutter_stripe.git
      path: packages/stripe

dependency_overrides:
  stripe_platform_interface:
    git:
      url: https://github.com/Flucadetena/flutter_stripe.git
      path: packages/stripe_platform_interface
```
